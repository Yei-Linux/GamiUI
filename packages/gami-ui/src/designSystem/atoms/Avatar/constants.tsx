export const options = {
  type: [] as const,
  zoomMode: ['outside', 'inside', 'none'] as const,
}

export const avatarProfilesImages = {
  one: 'https://i.imgur.com/vyW3w3r.png',
  two: 'https://i.imgur.com/lkIgoZP.png',
  three: 'https://i.imgur.com/fuFwKbQ.png',
  four: 'https://i.imgur.com/IXPZxIr.png',
  five: 'https://i.imgur.com/ub0bTXj.png',
}
