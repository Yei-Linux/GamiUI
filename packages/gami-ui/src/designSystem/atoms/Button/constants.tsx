export const options = {
  type: [
    'primary',
    'secondary',
    'tertiary',
    'cuaternary',
    'success',
    'warning',
    'danger',
    'info',
    'gradient',
  ] as const,
  typeHtml: ['reset','button','submit'] as const
}
