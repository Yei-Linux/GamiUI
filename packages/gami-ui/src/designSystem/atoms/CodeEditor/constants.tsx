export const ShowLineNumbersByDevice = {
  desktop: true,
  laptop: true,
  tablet: true,
  phone: false,
}

export const CodeCustomTheme = {
  base: 'vs' as const,
  inherit: false,
  colors: {
    'editor.background': '#1d2432',
    'editorCursor.foreground': '#536383',
    'editor.lineHighlightBackground': '#333e53',
    'editor.selectionBackground': '#536383',
    'editorActiveLineNumber.foreground': '#237893',
  },
  rules: [
    { token: '', foreground: '63b7fc', background: '1d2432' },
    { token: 'invalid', foreground: 'f44747' },
    { token: 'emphasis', fontStyle: 'italic' },
    { token: 'strong', fontStyle: 'bold' },

    { token: 'variable', foreground: '1AEBFF' },
    { token: 'variable.parameter', foreground: '9CDCFE' },
    { token: 'constant', foreground: 'ff636f' },
    { token: 'comment', foreground: '608B4E' },
    { token: 'number', foreground: '63b7fc' },
    { token: 'regexp', foreground: 'C0C0C0' },
    { token: 'annotation', foreground: 'ff636f' },
    { token: 'type', foreground: '3DC9B0' },

    { token: 'delimiter', foreground: 'dcdcdc' },
    { token: 'delimiter.html', foreground: 'dcdcdc' },

    { token: 'tag', foreground: 'ff636f' },
    { token: 'tag.id.pug', foreground: '4F76AC' },
    { token: 'tag.class.pug', foreground: '4F76AC' },
    { token: 'meta', foreground: 'D4D4D4' },
    { token: 'meta.tag', foreground: '9ddf6e' },
    { token: 'metatag', foreground: 'ff636f' },
    { token: 'metatag.content.html', foreground: '1AEBFF' },
    { token: 'metatag.html', foreground: 'ff636f' },
    { token: 'metatag.xml', foreground: 'ff636f' },
    { token: 'metatag.php', fontStyle: 'bold' },

    { token: 'key', foreground: '9CDCFE' },
    { token: 'string.key', foreground: '9CDCFE' },
    { token: 'string.value', foreground: '9ddf6e' },

    { token: 'attribute.name', foreground: 'ff636f' },
    { token: 'attribute.value', foreground: '3FF23F' },

    { token: 'string', foreground: '9ddf6e' },
    { token: 'string.sql', foreground: 'FF0000' },

    { token: 'keyword', foreground: 'ff636f' },
    { token: 'keyword.flow', foreground: 'C586C0' },

    { token: 'operator.sql', foreground: '778899' },
    { token: 'operator.swift', foreground: '909090' },
    { token: 'predefined.sql', foreground: 'FF00FF' },
  ],
}

export const codeOptions = (isShowLineNumbers: boolean) => {
  return {
    lineNumbers: isShowLineNumbers ? 'on' as const : 'off' as const,
    acceptSuggestionOnCommitCharacter: true,
    acceptSuggestionOnEnter: 'on' as const,
    accessibilitySupport: 'auto' as const,
    autoIndent: 'none' as const,
    automaticLayout: true,
    codeLens: true,
    colorDecorators: true,
    contextmenu: true,
    cursorBlinking: 'blink' as const,
    cursorSmoothCaretAnimation: false,
    cursorStyle: 'line' as const,
    disableLayerHinting: false,
    disableMonospaceOptimizations: false,
    dragAndDrop: false,
    fixedOverflowWidgets: false,
    folding: true,
    foldingStrategy: 'auto' as const,
    fontLigatures: false,
    formatOnPaste: false,
    formatOnType: false,
    hideCursorInOverviewRuler: false,
    highlightActiveIndentGuide: true,
    links: true,
    mouseWheelZoom: false,
    multiCursorMergeOverlapping: true,
    multiCursorModifier: 'alt' as const,
    overviewRulerBorder: true,
    overviewRulerLanes: 2,
    quickSuggestions: true,
    quickSuggestionsDelay: 100,
    readOnly: false,
    renderControlCharacters: false,
    renderFinalNewline: true,
    renderIndentGuides: true,
    renderLineHighlight: 'all' as const,
    renderWhitespace: 'none' as const,
    revealHorizontalRightPadding: 30,
    roundedSelection: true,
    rulers: [],
    scrollBeyondLastColumn: 5,
    scrollBeyondLastLine: true,
    selectOnLineNumbers: true,
    selectionClipboard: true,
    selectionHighlight: true,
    showFoldingControls: 'mouseover' as const,
    smoothScrolling: false,
    suggestOnTriggerCharacters: true,
    wordBasedSuggestions: true,
    wordSeparators: `~!@#$%^&*()-=+[{]}|;:'",.<>/?`,
    wordWrap: 'off' as const,
    wordWrapBreakAfterCharacters: '\t})]?|&,;',
    wordWrapBreakBeforeCharacters: '{([+',
    wordWrapBreakObtrusiveCharacters: '.',
    wordWrapColumn: 80,
    wordWrapMinified: true,
    wrappingIndent: 'none' as const,
  }
}

export const codeTheme = 'myTheme'

export const options = {
  type: [] as const,
}
