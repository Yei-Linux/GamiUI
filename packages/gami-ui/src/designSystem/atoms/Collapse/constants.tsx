import React from 'react'

import Title from '../Title'

export const options = {
  type: [] as const,
}

export const TitleCollapses = [
  <Title key="1" level="h3">
    Title A
  </Title>,
  <Title key="2" level="h3">
    Title B
  </Title>,
  <Title key="3" level="h3">
    Title C
  </Title>,
]
