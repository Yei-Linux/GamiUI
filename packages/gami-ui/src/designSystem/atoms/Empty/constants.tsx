export const iconsEmpty = {
  base: 'empty',
  sad: 'sad',
}

export const options = {
  type: ['base', 'sad'] as const,
}
