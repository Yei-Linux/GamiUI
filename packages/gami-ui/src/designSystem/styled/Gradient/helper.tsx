export const makeGradient = (firstColor: string, secondColor: string) =>
  `linear-gradient(to right, ${firstColor}, ${secondColor}`
