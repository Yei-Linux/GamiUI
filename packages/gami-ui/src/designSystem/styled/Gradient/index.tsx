export { default } from './Gradient'
export { options } from './constants'
