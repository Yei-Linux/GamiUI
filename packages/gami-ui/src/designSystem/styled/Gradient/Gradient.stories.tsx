import React from 'react'
import { Meta } from '@storybook/react'

import Gradient from '.'
import { options as optionsGradient } from './constants'
import Container from 'designSystem/layouts/Container'

export default {
  title: 'Styled/Gradient',
  component: Gradient,
  argTypes: {
    name: { control: { type: 'select', options: optionsGradient.names } },
  },
} as Meta

export const Basic = (args: any) => (
  <div
    style={{
      width: '100%',
      background:
        'url(https://res.cloudinary.com/edapp/images/w_680,h_350,c_fill,g_auto/v1615782673/wordpress/production/staff-learning-with-gamification/staff-learning-with-gamification-680x350.jpg) no-repeat center center',
      backgroundSize: 'cover',
      display: 'flex',
      justifyContent: 'center',
      padding: '1rem',
      flexWrap: 'wrap',
    }}
  >
    <Gradient {...args}>
      <Container style={{ width: '100%' }}>Gami Gradient</Container>
    </Gradient>
  </div>
)
Basic.args = {}
