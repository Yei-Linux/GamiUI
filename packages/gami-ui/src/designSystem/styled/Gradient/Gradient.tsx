import React, { Fragment } from 'react'
import { GradientsNames } from 'core/domain/types'
import useCloneElement from 'hooks/useCloneElements'
import { GradientTypes } from './constants'
import { makeGradient } from './helper'

interface CustomGradientColors {
  first: string
  second: string
}

export interface GradientProps {
  /**
   * Component will be modified
   */
  children: React.ReactNode
  /**
   * Gradient Name
   */
  name: GradientsNames
  /**
   * Gradient custom colors in case not use gradient name
   */
  colors?: CustomGradientColors | null
}

const Gradient = ({
  children,
  name = 'almost',
  colors = null,
}: GradientProps) => {
  const makeStyleGradient = () =>
    !colors
      ? GradientTypes[name]
      : { background: makeGradient(colors?.first, colors?.second) }

  const { validatorChildrenLength, childrenWithProps } = useCloneElement({
    children,
    propsElement: {
      style: makeStyleGradient(),
    },
    maxChildrenNumber: 1,
    childrenTypes: [],
  })

  if (validatorChildrenLength(childrenWithProps)) {
    return null
  }

  return <Fragment>{childrenWithProps}</Fragment>
}

export default Gradient
