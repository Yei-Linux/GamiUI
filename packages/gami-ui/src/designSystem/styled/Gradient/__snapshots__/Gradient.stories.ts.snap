// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Styled/Gradient Basic 1`] = `
<div
  style={
    Object {
      "background": "url(https://res.cloudinary.com/edapp/images/w_680,h_350,c_fill,g_auto/v1615782673/wordpress/production/staff-learning-with-gamification/staff-learning-with-gamification-680x350.jpg) no-repeat center center",
      "backgroundSize": "cover",
      "display": "flex",
      "flexWrap": "wrap",
      "justifyContent": "center",
      "padding": "1rem",
      "width": "100%",
    }
  }
>
  <div
    className="css-1ohc8tn-CardWrapper e1ft0i6y0"
    height="NORMAL"
    style={
      Object {
        "background": "linear-gradient(to right, rgb(221, 214, 243), rgb(250, 172, 168))",
      }
    }
    width="NORMAL"
  >
    <p
      style={
        Object {
          "padding": "1rem",
        }
      }
    >
      Hola como estas
    </p>
  </div>
</div>
`;
